locator/tablets.cc

```diff
@@ -670,9 +670,23 @@ future<bool> check_tablet_replica_shards(const tablet_metadata& tm, host_id this
 }
 
 class tablet_effective_replication_map : public effective_replication_map {
+
+public:
+    using replication_factor_index_t = uint16_t;
+    using replication_factor_list = utils::small_vector<replication_factor_index_t, 8>;
+
+private:
+
     table_id _table;
     tablet_sharder _sharder;
     mutable const tablet_map* _tmap = nullptr;
+    // The `_datacenter_map` stores the association between each data center
+    // and its index in the `replication_factor_list` in `_tablet_to_dc_replication_factor_map`.
+    const absl::flat_hash_map<sstring, replication_factor_index_t> _datacenter_map;
```

Comment:
I think naming it `dc_index_type` would be clearer. The datacenter_map does not depend on the replication factor map, it's the other way around. It could be used for other things than replication factor. The value represents a dc, so it's more appropriate to name it dc_index or dc_id. The size of the type depends on how many DCs there can be.